### PR TITLE
Add transaction hash for pending/success/error messages

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: universal-login-relayer/lib/core/services/messages/QueueService.ts [hunk 1]
```diff
@@ -27,11 +28,14 @@ class QueueService {
   }
 
   async execute(messageHash: string) {
-    await this.messageRepository.setMessageState(messageHash, 'Pending');
     try {
       const signedMessage = await this.messageRepository.getMessage(messageHash);
-      const {hash} = await this.messageExecutor.executeAndWait(signedMessage);
-      await this.messageRepository.markAsSuccess(messageHash, hash!);
+      const transactionResponse = await this.messageExecutor.execute(signedMessage);
```
Comment (JustynaBroniszewska) [inline universal-login-relayer/lib/core/services/messages/QueueService.ts]:
```ts
    const {hash, wait} = await this.messageExecutor.execute(signedMessage);
    await wait();
```

File: docs/source/overview.rst [hunk 2]
```diff
@@ -134,8 +134,8 @@ The message starts it's journey when it is created and signed by the user (i.e.
 
 - **await signature** ``optional``- Relayer waits to collect all required signatures if the message requires more than one signature.
 - **queued** - Message is queued to be sent.
-- **pending** - Message is propagated to the network and waits to be mined.
-- **sucess** / **error** - Mined transaction is a success or an error. In a success state, the message has a transaction hash. In an error state, the message has an error message.
+- **pending** - Message is propagated to the network and waits to be mined. In a pending state, the message has a transaction hash.
+- **sucess** / **error** - Mined transaction is a success or an error. In a success state, the content of message status is not changed. In an error state, the message has an error message.
```
Comment (marekkirejczyk) [inline docs/source/overview.rst]: In both success and error states Message contains transaction hash.

File: universal-login-relayer/test/integration/core/services/MultiSignatureExecute.js [hunk 3]
```diff
@@ -56,9 +56,12 @@ describe('INT: MultiSignatureExecute', async () => {
       const signedMessage0 = createSignedMessage(msg, wallet.privateKey);
       const signedMessage1 = createSignedMessage(msg, actionKey);
       await messageHandler.handleMessage(signedMessage0);
-      await messageHandler.handleMessage(signedMessage1);
+      const {messageHash} = await messageHandler.handleMessage(signedMessage1);
       await messageHandler.stopLater();
       expect(await provider.getBalance(msg.to)).to.eq(expectedBalance);
+      const {state, transactionHash} = await messageHandler.getStatus(messageHash);
+      expect(transactionHash).to.not.be.null;
```
Comment (marekkirejczyk) [inline universal-login-relayer/test/integration/core/services/MultiSignatureExecute.js]: Please something more strict, at least:
`expect(transactionHash).to.be.properHex(...);`

File: universal-login-relayer/lib/integration/ethereum/MessageExecutor.ts [hunk 4]
```diff
@@ -9,13 +9,15 @@ export class MessageExecutor {
...
     const transactionReq: providers.TransactionRequest = messageToTransaction(signedMessage);
     await this.messageValidator.validate(signedMessage, transactionReq);
-    const transactionRes: providers.TransactionResponse = await this.wallet.sendTransaction(transactionReq);
-    await this.wallet.provider.waitForTransaction(transactionRes.hash!);
-    await this.onTransactionSent(transactionRes);
-    return transactionRes;
+    return this.wallet.sendTransaction(transactionReq);
+  }
+
+  async waitAndHandleTransaction(transactionResponse: providers.TransactionResponse) {
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/integration/ethereum/MessageExecutor.ts]: We don't need that method anymore, we can inline it.

File: universal-login-relayer/lib/core/services/messages/QueueService.ts [hunk 5]
```diff
@@ -27,11 +28,14 @@ class QueueService {
...
+      const {hash} = transactionResponse;
+      ensureNotNull(hash, TransactionHashNotFound);
+      await this.messageRepository.markAsPending(messageHash, hash!);
+      await this.messageExecutor.waitAndHandleTransaction(transactionResponse);
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/core/services/messages/QueueService.ts]: We should move OnTransactionSent to Queue form executor and inline waiting to this method. No reason to push it down the abstract later.

**Pull request merged**